doc-checking-app-api/src/main/java/uk/gov/di/authentication/app/lambda/DocAppCallbackHandler.java

```diff
@@ -111,6 +119,32 @@ public APIGatewayProxyResponseEvent docAppCallbackRequestHandler(
             APIGatewayProxyRequestEvent input, Context context) {
         LOG.info("Request received to DocAppCallbackHandler");
         try {
+            if (isCustomDocAppClaimEnabledAndCustomErrorPresent(input.getQueryStringParameters())) {
+                var clientId = configurationService.getDocAppRPClientId();
+                var clientRegistry =
+                        dynamoClientService
+                                .getClient(clientId)
+                                .orElseThrow(
+                                        () ->
+                                                new InvalidClientException(
+                                                        format(
+                                                                "No client found with clientId: %s",
+                                                                clientId)));
+                var redirectUrl = clientRegistry.getRedirectUrls().get(0);
+                //        TODO - Pass error received from DCMAW back to RP and ensure additional
+                // query params are included
+                var errorResponse =
+                        new AuthenticationErrorResponse(
+                                URI.create(redirectUrl),
```

Comment:
I'm pretty confident that we don't want to use "temporarily unavailable", let's review this Thursday :-)

Reply:
We don't. That's why there is a TODO to update it when the expected error is known

Reply:
We have an [RFC update](https://github.com/alphagov/digital-identity-architecture/pull/158) that I'd appreciate your and @SenthilKKumar eyes on. This RFC really only covers credential issuers and IPV Core, so potentially we should have a similar(ish) update in the OpenID Connect profile RFC too?

LMK if you'd like to discuss

Reply:
@galund mind having a look? Error has been updated for to reflect the RFC. There will be a slightly different implementation with IPV to be able to get the session from the state.

Reply:
So the goal (as far as I'm concerned) of our conversations was to ensure Orchestration didn't have to do something particularly different for DCMAW vs IPV Core. Sounds like we need a rerun of the conversation the other day as it feels like we haven't arrived in that place.

I thought we agreed that the trigger will be the `access_denied` and you should *not* be parsing the `error_description` looking for a special "missing context" string.

Instead, when you receive the OAuth error, there are two cases depending whether there is a current active valid session in Orchestration:

* if there is a session, the user cancelled out normally.
* if there is no session, then the user may be able to pick up their journey if they log in again and come back.

We can return the same error in both cases to HMRC ie `access_denied`, and we should use the same `state` that was passed in on the original request. Note that `state` is mandatory, so I don't think we can get away with not looking up the session, in that second case?

HMRC should then perform similar logic:

* if there is a session, the user cancelled out and they should find another way to prove the user's identity (or whatever they do next)... but they might also want to *offer* the user to try again with the app, because people change their minds.
* if there is no session, they will need to sign the user in with SCP again, and *then* they should offer the user to try again with the App (and in this case, they are likely - we hope - to succeed).

Reply:
Currently there is a technical challenge, Orchestration will be able to identify appropriate RP session using `state` only when the RP sessions are managed using DynamoDB instead of Redis. 

cc: @mrwilson

Reply:
Ah, so maybe we can't fix this for anyone but HMRC just yet anyway. Still it'd be nice to be clear with HMRC that what they should do depends on whether they have a current session, rather than on magic words in the error description (assuming that really is what we agreed the other day, and I think you had to miss that Senthil but if you and Alex and anyone else wants to chat, do shout.)

Reply:
Senthil and I spoke. So we are not dependent on the move to Dynamo, we are going to store the client session id against the state in Redis before we hand off to DCMAW. This is outlined in PR https://github.com/alphagov/di-authentication-api/pull/2793 which supersedes this one